infra/modular-examples/aws-msft-365/main.tf

```diff
@@ -74,34 +81,33 @@ module "msft-connection" {
   required_oauth2_permission_scopes = each.value.required_oauth2_permission_scopes
 }
 
-module "msft-connection-auth" {
-  for_each = module.worklytics_connector_specs.enabled_msft_365_connectors
-
-  source = "../../modules/azuread-local-cert"
-  # source = "git::https://github.com/worklytics/psoxy//infra/modules/azuread-local-cert?ref=v0.4.10"
+module "cognito-identity" {
+  source = "../../modules/aws-cognito-identity-cli"
 
-  application_object_id = module.msft-connection[each.key].connector.id
-  rotation_days         = 60
-  cert_expiration_days  = 180
-  certificate_subject   = var.certificate_subject
+  identity_pool_id = module.cognito-identity-pool.pool_id
+  aws_region       = var.aws_region
+  login-ids        = { for k in keys(module.msft-connection) : k => "${module.cognito-identity-pool.developer_provider_name}=${module.msft-connection[k].connector.application_id}" }
+  aws_role         = var.aws_assume_role_arn
 }
 
-module "msft-365-connector-key-secrets" {
+module "msft-connection-auth-federation" {
   for_each = module.worklytics_connector_specs.enabled_msft_365_connectors
 
-  source = "../../modules/private-key-aws-parameter"
-  # source = "git::https://github.com/worklytics/psoxy//infra/modules/private-key-aws-parameter?ref=v0.4.10"
+  source = "../../modules/azuread-federated-credentials"
+  # source = "git::https://github.com/worklytics/psoxy//infra/modules/azuread-federated-credentials?ref=v0.4.11"
 
-  instance_id = each.key
+  application_object_id = module.msft-connection[each.key].connector.id
+  display_name          = "AccessFromAWS"
+  description           = "AWS federation to be used for psoxy Connector - ${each.value.display_name}${var.connector_display_name_suffix}"
+  issuer                = "https://cognito-identity.amazonaws.com"
+  audience              = module.cognito-identity-pool.pool_id
+  subject               = module.cognito-identity.identity_id[each.key]
 
-  private_key_id = module.msft-connection-auth[each.key].private_key_id
-  private_key    = module.msft-connection-auth[each.key].private_key
-  ssm_path       = var.aws_ssm_param_root_path
-  kms_key_id     = var.aws_ssm_key_id
+  depends_on = [module.cognito-identity]
```

Comment:
i would hope TF smart enough that this explicit dependency isn't needed; while maybe `pool_id` known at the time of plan, I would think `identity_id` may not be known until that `cognito-identity` module is applied, right?